### ISSUE TITLE
baselayerchange event fires only after the second time base layer is changed

### ISSUE THREAD

**Issue by charlesfrys** [Leaflet/Leaflet#6737]
Seems like a bug to me. I read something in a forum saying that this was because the event was fired only when a layer was added which is the case only after the second change.

https://gis.stackexchange.com/questions/259894/why-in-leaflet-does-baselayerchange-event-not-fire-until-the-second-time-the-bas

If you could solve this it would be great.

**Comment by IvanSanchez** [comment]
Behaviour as expected, as explained at https://gis.stackexchange.com/a/259909 . 

Otherwise, duplicate of #4925.
